src/priority-queue.ts

```diff
@@ -0,0 +1,262 @@
+import {ArmorCollection} from './collection';
+import {ArmorCollectionSelector} from './selector';
+
+/**
+ * decide what to do when no comparator is passed
+ */
+
+export interface ArmorPriorityQueueOptions<T> {
+	elements?: Array<T>;
+	comparator?: (nodeOneIndex: number, nodeTwoIndex: number) => boolean;
+}
+
+const defaultComparator = function (nodeOneIndex: number, nodeTwoIndex: number) {
+	if (typeof nodeOneIndex !== 'number') {
+		return false;
+	}
+	if (typeof nodeTwoIndex !== 'number') {
+		return false;
+	}
+
+	return JSON.stringify(this.elements[nodeOneIndex]) < JSON.stringify(this.elements[nodeTwoIndex]);
+};
+
+export class ArmorPriorityQueue<T> implements ArmorCollection<T> {
+	public elements: T[];
+	public comparator: (nodeOneIndex: number, nodeTwoIndex: number) => boolean;
+
+	constructor(options: ArmorPriorityQueueOptions<T> = {}) {
```

Comment:
- [ ] Use the `?` operator to make options optional instead of defaulting to empty object.
```typescript
constructor(options?: ArmorPriorityQueueOptions<T>) {
  ...
}
```

Using a method to parse optional properties from an options/config object is helpful for testing. Testing whether the options object was passed in at all is an easy way to early drop from that method. To do that, the default case for options is undefined. 
Because the default is an empty object, the following is true:
```typescript
const options = {};

// empty objects are truthy.
// This will print 'options true' in the console.
if (options) {
   console.log('options true');
}
```